apps/sports-helsinki/src/domain/search/combinedSearch/hooks.ts

```diff
@@ -0,0 +1,163 @@
+import { PARAM_SEARCH_TYPE } from './constants';
+import type { SearchTabId } from './searchTabs/tabsContext';
+import { isSearchTabId, useTabsContext } from './searchTabs/tabsContext';
+
+/**
+ * Use an URL parameter to scroll to
+ * the previously used search result card on a page change.
+ */
+export const useScrollToSearchResultItem = () => {
+  const router = useRouter();
+  const scrollTo = router.query?.scrollTo;
+  const listRef = useRef<HTMLUListElement | null>(null);
+
+  useEffect(() => {
+    const listElement = listRef.current;
+
+    if (scrollTo) {
+      const listItemElement = listElement?.querySelector(
+        decodeURIComponent(scrollTo.toString())
+      );
+
+      if (listItemElement) {
+        listItemElement.scrollIntoView({
+          block: 'center',
+        });
+      }
+    }
+  }, [scrollTo]);
+};
+
+export const useSearchTabsWithParams = (defaultTab: SearchTabId) => {
+  const router = useRouter();
+  const searchParams = new URLSearchParams(router.asPath.split('?')[1]);
+  const searchTypeParam = searchParams.get(PARAM_SEARCH_TYPE);
+  const initTab = searchTypeParam
+    ? (searchTypeParam as SearchTabId)
+    : defaultTab;
+
+  // If the search type param is not given in the URL, set it there.
+  React.useEffect(() => {
+    if (!isSearchTabId(searchTypeParam)) {
+      router.push(
+        { query: { ...router.query, [PARAM_SEARCH_TYPE]: defaultTab } },
+        undefined,
+        { shallow: true }
+      );
+    }
+  }, [searchTypeParam, router, defaultTab]);
+
+  return { initTab, searchTypeParam };
+};
+
+export function useLazyVenueSearchForTabCount() {
+  const { setResultCount } = useTabsContext();
+  const venueSearchFilters = useUnifiedSearchVariables();
+  const variables = React.useMemo(
+    () => ({
+      ...venueSearchFilters,
+      includeHaukiFields: AppConfig.isHaukiEnabled,
+    }),
+    [venueSearchFilters]
+  );
+  const [search, { loading, data, ...delegatedProps }] = useLazyQuery(
+    SearchListDocument,
+    {
+      variables,
+      // FIXME: Set the fetch policy to not trigger cache,
+      // or the Apollo client fails to request
+      // same query multiple times. Depending on settings,
+      // the client stops the query when the first response is given,
+      // and uses the cache for the next query with same name or
+      // returns the items twice.
+      // Since the query is exactly the same as the actual listing query,
+      // it would be really nice to have the result in cache already.
+      // See https://community.apollographql.com/t/executing-the-same-query-multiple-times-with-different-variables/3052
+      fetchPolicy: 'no-cache',
+      ssr: false,
+    }
+  );
+
+  React.useEffect(() => {
+    const count = () => {
+      if (loading || !data) {
+        return null;
+      }
+      return (data as SearchListQuery)?.unifiedSearch?.count ?? 0;
+    };
+    setResultCount('Venue', count());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
At least the setResultCount was left out in a purpose, but I had an issue there earlier. It could work now. Earlier it changed in a never ending loop, but it might be OK now when the setter is fixed inside the TabsContext.

Reply:
no, it still does not work if I add the `setResultCount` as a dependency for the hook.

> next-dev.js?36dd:20 Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.